fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    this.getAllTodos();
+  }
+
+  getAllTodos() {
+    fetch('/api/todos')
+      .then(res => res.json())
+      .then(data => this.setState({
+        todos: data
+      }))
+  }
+
+  addTodo(newTodo) {
+    const cloneState = [...this.state.todos]
+    cloneState.push(newTodo)
```

Comment:
You don't want to push in `newTodo` into the copy here, it doesn't have an id yet. The id is generated from the database.